lib/youtube/youtube.ex

```diff
@@ -7,45 +7,43 @@ defmodule Youtube do
   plug(Tesla.Middleware.BaseUrl, @base_url)
   plug(Youtube.OAuth.Middleware)
 
-  def get_channel_videos(channel_id) do
-    %{status: 200, body: body} =
-      get!("/search",
-        query: [
-          channelId: channel_id,
-          order: "date",
-          type: "video",
-          part: "snippet"
-        ]
-      )
-
+  def get_channel_last_video(channel_id) do
+    %{status: 200, body: body} = fetch_videos(channel_id)
     %{"items" => items} = body
 
-    now = DateTime.utc_now()
+    videos =
+      Enum.map(items, fn item ->
+        %{
+          video_id: item["id"]["videoId"],
+          video_title: item["snippet"]["title"],
+          published_time: DateTime.from_iso8601(item["snippet"]["publishedAt"])
+        }
+      end)
+
     # Only keep videos published the last 6 days
-    find_last_videos(items, DateTime.add(now, -6 * 24 * 3600, :second)) |> last_video()
+    now = DateTime.utc_now()
+    [last_video] = videos_after_timestamp(videos, DateTime.add(now, -6 * 24 * 3600, :second))
```

Comment:
Just FYI: in Elixir 1.14, [`DateTime.add`](https://hexdocs.pm/elixir/DateTime.html#add/4) supports `:day` as well